src/context/toast-context.js

```diff
@@ -0,0 +1,36 @@
+import { createContext, useContext, useState, useEffect } from "react";
+
+const initialValue = {
+  type: "",
+  message: "",
+};
+
+const ToastContext = createContext(initialValue);
+
+const ToastProvider = ({ children }) => {
+  const [toastMessage, setToastMessage] = useState(initialValue);
+  const [timerId, setTimerId] = useState(null);
+
+  useEffect(() => {
+    clearTimeout(timerId);
```

Comment:
suggestion: use cleanup function, as its meant for cleaning things

Reply:
Update with cleanup function